internal/host/validator.go

```diff
@@ -921,67 +766,57 @@ func (v *validator) missingNTPSyncResult(db *gorm.DB, host *models.Host) Validat
 	return ValidationFailure
 }
 
-func (v *validator) isNTPSynced(c *validationContext) ValidationStatus {
+func (v *validator) isNTPSynced(c *validationContext) (ValidationStatus, string) {
+
+	var status ValidationStatus
+	var message string
+
 	var sources []*models.NtpSource
 
 	if c.host.NtpSources == "" {
-		return v.missingNTPSyncResult(c.db, c.host)
-	}
-
-	if err := json.Unmarshal([]byte(c.host.NtpSources), &sources); err != nil {
+		status = v.missingNTPSyncResult(c.db, c.host)
+	} else if err := json.Unmarshal([]byte(c.host.NtpSources), &sources); err != nil {
 		v.log.WithError(err).Warn("Parse NTP sources")
-		return ValidationError
-	}
-
-	for _, source := range sources {
-		if source.SourceState == models.SourceStateSynced {
-			return ValidationSuccess
+		status = ValidationError
+	} else {
+		status = v.missingNTPSyncResult(c.db, c.host)
+		for _, source := range sources {
+			if source.SourceState == models.SourceStateSynced {
+				status = ValidationSuccess
+			}
 		}
 	}
 
-	return v.missingNTPSyncResult(c.db, c.host)
-}
-
-func (v *validator) printNTPSynced(c *validationContext, status ValidationStatus) string {
 	switch status {
 	case ValidationSuccess:
-		return "Host NTP is synced"
+		message = "Host NTP is synced"
 	case ValidationFailure:
-		return "Host couldn't synchronize with any NTP server"
+		message = "Host couldn't synchronize with any NTP server"
 	case ValidationError:
-		return "Parse error for NTP sources"
+		message = "Parse error for NTP sources"
 	default:
-		return fmt.Sprintf("Unexpected status %s", status)
+		message = fmt.Sprintf("Unexpected status %s", status)
 	}
+
+	return status, message
 }
 
-func (v *validator) sucessfullOrUnknownContainerImagesAvailability(c *validationContext) ValidationStatus {
+func (v *validator) sucessfullOrUnknownContainerImagesAvailability(c *validationContext) (ValidationStatus, string) {
 	imageStatuses, err := common.UnmarshalImageStatuses(c.host.ImagesStatus)
 	if err != nil {
 		v.log.WithError(err).Warn("Parse container image statuses")
-		return ValidationError
+		return ValidationError, "Validation error"
 	}
-
-	return boolValue(allImagesValid(imageStatuses))
-}
-
-func (v *validator) printSucessfullOrUnknownContainerImagesAvailability(c *validationContext, status ValidationStatus) string {
-	switch status {
-	case ValidationSuccess:
-		return "All required container images were either pulled successfully or no attempt was made to pull them"
-	case ValidationFailure:
+	if boolValue(allImagesValid(imageStatuses)) == ValidationFailure {
```

Comment:
`allImagesValid` returns a bool, just use that in the condition. Why convert and compare to a different type?